File: code/SoundPlayground/SoundPlayground/Parser/AbstractSyntaxTree/ContextModifiers/VelocityModifierNode.cs
using System.Collections.Generic;
using SoundPlayground.Core;

namespace SoundPlayground.Parser.AbstractSyntaxTree.ContextModifiers {
    public class VelocityModifierNode : ContextModifierNode {
        public int Velocity { get; set; }

        public VelocityModifierNode ( int velocity ) {
            Velocity = velocity;
        }

        public override void Modify ( Context context ) {
            context.Velocity = Velocity;
        }
    }
}

File: code/SoundPlaygroundPy/parser/abstract_syntax_tree/context_modifiers/velocity_modifier_node.py
from .context_modifier_node import ContextModifierNode
from core.events import ContextChangeEvent
from core import Context

class VelocityModifierNode( ContextModifierNode ):
    def __init__ ( self, velocity ):
        super().__init__()

        self.velocity = velocity

    def modify ( self, context : Context ):
        context.velocity = self.velocity;

        yield ContextChangeEvent( context.cursor, "velocity", context.velocity )
